### ISSUE TITLE
Forms. Submitted state

### ISSUE THREAD

**Issue by kamarouski** [angular/angular#12300]
It'd be nice to have .submitted : bool property of FormGroup as in Angular 1

It's useful while performing validation and  we want to display the error(s) only after user tried to submit the form.

e.g. We don't want to display 'Invalid Email Format' error when user typing email for the first time. Only after user submitted the form or field lost focus.  valid || pristine don't help here.

Additionally, just a question. If the form is invalid why do we perform ngSubmit? Form is not ready to be submitted.

* **Angular version:** 2.1.X

**Comment by pkozlowski-opensource** [comment]
> valid || pristine don't help here.

How about `touched` that will be valid only when a control looses focus?

**Comment by kara** [comment]
@kamarouski  We do actually have a submitted property, but it exists on the form directives rather than the `FormGroup`. This is because it only really makes sense on the top level, and `FormGroups` can be nested. Ex:

**template-driven**
```html
<form #f="ngForm">
   <div *ngIf="f.submitted && f.invalid">Invalid email.</div>
   <input name="email" ngModel>
</form>
```

**reactive**
```ts
@ViewChild(FormGroupDirective) formGroupDir: FormGroupDirective;

get showError() {
   return this.form.invalid && this.formGroupDir.submitted;
}
```

Is that what you're looking for?

**Comment by kamarouski** [comment]
Thanks @kara . That's exactly what I'm looking for

**Comment by kara** [comment]
Excellent! In that case, closing the issue.
